rpc/grpc/server/server.go

```diff
@@ -48,10 +49,7 @@ func newServerBuilder(listener net.Listener) *serverBuilder {
 func Listen(addr string) (net.Listener, error) {
 	parts := strings.SplitN(addr, "://", 2)
 	if len(parts) != 2 {
-		return nil, fmt.Errorf(
-			"invalid listening address %s (use fully formed addresses, including the tcp:// or unix:// prefix)",
-			addr,
-		)
+		return nil, jsonrpcerrors.ErrInvalidRemoteAddress{Addr: addr}
```

Comment:
Could we not do lateral imports here? It's OK to duplicate the odd error for the gRPC case. Or move it to the top-level `rpc` package, now that you have created it to collect error cases that are reused between subdirectory packages.

Reply:
I think moving errors to the top-level RPC package makes sense. This way, we keep things clean and avoid duplication. Does that sound good to you?

Reply:
Yes, maybe not for all errors, but those that are reused between different packages.
The alternative is to simply duplicate these errors in each package. It seems like we have only a few cases where the same error messages are used in different packages, so maybe we should not commit to a separate package API just for these few cases.

Reply:
Yeah make sense. I have duplicated the error in both packages.